jetty-webapp/src/main/java/org/eclipse/jetty/webapp/ClasspathPattern.java

```diff
@@ -685,21 +686,7 @@ public boolean match(Class<?> clazz)
     {       
         try
         {
-            Boolean byName = _patterns.isIncludedAndNotExcluded(clazz.getName());
-            if (Boolean.FALSE.equals(byName))
-                return byName; // Already excluded so no need to check location.
-            URI location = TypeUtil.getLocationOfClass(clazz);
-            Boolean byLocation = location == null ? null
-                    : _locations.isIncludedAndNotExcluded(location);
-            
-            if (LOG.isDebugEnabled())
-                LOG.debug("match {} from {} byName={} byLocation={} in {}",clazz,location,byName,byLocation,this);
-            
-            // Combine the tri-state match of both IncludeExclude Sets
-            boolean included = Boolean.TRUE.equals(byName) || Boolean.TRUE.equals(byLocation)
-                || (byName==null && !_patterns.hasIncludes() && byLocation==null && !_locations.hasIncludes());
-            boolean excluded = Boolean.FALSE.equals(byName) || Boolean.FALSE.equals(byLocation);
-            return included && !excluded;
+            return IncludeExcludeSet.or(_patterns, clazz.getName(), _locations, ()->TypeUtil.getLocationOfClass(clazz));
```

Comment:
this syntax is really strange to me.
this doesn't look like 2 IncludeExclude sets being or'd together.

Reply:
I guess just "or" is the wrong name.    The operation applies to item x, set of X, item y and set of Y.  We want to return true if x is in setX OR y is in setY.   Elsewhere we need to do true if x is in setX AND y is in setY.
This is not helped by some old field names.... let me fix those first and then ponder if there is better name and/or syntax for these operations....

Reply:
I've improved the javadoc on those methods... which might help a little.
I'm struggling to find two names for those methods that are no "or" and "and", because that is the fundamental difference between them.
The only other alternative I can think of is just inline them and delete the methods?

Reply:
The or case could be existsInAny() with the and case being existsInAll()?